src/go/rpk/pkg/cli/cmd/wasm/list.go

```diff
@@ -0,0 +1,170 @@
+// by the Apache License, Version 2.0
+
+package wasm
+
+import (
+	"encoding/json"
+	"strconv"
+	"strings"
+
+	"github.com/Shopify/sarama"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/cli/ui"
+	"github.com/vectorizedio/redpanda/src/go/rpk/pkg/kafka"
+)
+
+type coprocInfo struct {
+	InputTopics []string `json:"input_topics"`
+	Description string   `json:"description"`
+}
+
+type coprocNodeInfo struct {
+	NodeId       int                   `json:"node_id"`
+	Status       string                `json:"status"`
+	Coprocessors map[string]coprocInfo `json:"coprocessors"`
+}
+
+func NewListCommand(
+	clientClosure func() (sarama.Client, error),
+	consumerClosure func(sarama.Client) (sarama.Consumer, error),
+) *cobra.Command {
+	var pretty bool
+
+	command := &cobra.Command{
+		Use:   "list",
+		Short: "display the current status of functions within redpanda",
+		RunE: func(_ *cobra.Command, args []string) error {
+			return wasmList(clientClosure, consumerClosure, pretty)
+		},
+	}
+
+	command.Flags().BoolVar(&pretty, "raw", false, "Print the raw json to stdout")
+	return command
+}
+
+func printResults(dataSet *map[int64]coprocNodeInfo, raw bool) {
+	if raw {
+		stringRep, _ := json.Marshal(dataSet)
+		log.Info(string(stringRep))
+		return
+	}
+	t := ui.NewRpkTable(log.StandardLogger().Out)
+	t.SetColWidth(80)
+	t.SetAutoWrapText(false)
+	header := [...]string{"Script Name", "Node ID", "Input Topics", "Description"}
+	t.SetHeader(header[:])
+	for _, val := range *dataSet {
+		for name, data := range val.Coprocessors {
+			topics := strings.Join(data.InputTopics, ",")
+			topics = "[" + topics + "]"
+			row := [...]string{name, strconv.Itoa(val.NodeId), topics, data.Description}
+			t.Append(row[:])
+		}
+	}
+	t.Render()
+	// TODO: Print all down status updates as well
+	// log.Infof("Wasm Engine Status: %s", val.Status)
+}
+
+func accrueResults(
+	messages []*sarama.ConsumerMessage,
+) (*map[int64]coprocNodeInfo, error) {
+	allData := make(map[int64]coprocNodeInfo)
+	for _, msg := range messages {
+		msgKey, _ := strconv.ParseInt(string(msg.Key), 10, 64)
+		var info coprocNodeInfo
+		info.Coprocessors = make(map[string]coprocInfo)
+		if err := json.Unmarshal(msg.Value, &info); err != nil {
+			log.Error("Failed to deserialize json, error within redpanda")
+			return nil, err
+		}
+		allData[msgKey] = info
+	}
+	return &allData, nil
+}
+
+func consumeStatusUpdates(
+	pc sarama.PartitionConsumer, finalOffset int64,
+) []*sarama.ConsumerMessage {
+	var messages []*sarama.ConsumerMessage
+	doneCh := make(chan struct{})
+	go func() {
+		for {
+			select {
+			case msg := <-pc.Messages():
+				messages = append(messages, msg)
+				if msg.Offset >= (finalOffset - 1) {
+					doneCh <- struct{}{}
+				}
+			case err := <-pc.Errors():
+				if err != nil {
+					log.Errorf(
+						"Got an error consuming topic '%s', partition %d: %v",
```

Comment:
I think this could be confusing to the user. As far as they're concerned, they're listing the wasm scripts, not consuming from a topic (even if that's what actually happens behind the scene). This could be logged at debug level, and another, friendlier message could be logged at info instead.